File: scala/src/main/scala/recover.scala
import com.jambit.practicalresult.*

import scala.util.{Left, Right}

@main def recover: Unit =
  val result: Either[String, List[Int]] = listService(-1)

  println(result)

  val recovered: Either[String, List[Int]] = result match {
    case Left(a) => Right(List())
    case Right(b) => result
  }

  println(recovered)

File: kotlin/src/main/java/Recover.kt
import arrow.core.Either

fun main() {
  val result: Either<String, List<Int>> = listService(-1)

  println(result)

  val recovered: Either<String, List<Int>> = when (result) {
    is Either.Right -> result
    is Either.Left  -> Either.Right(emptyList())
  }

  println(recovered)
}